gap/attributes/attr.gi

```diff
@@ -777,10 +777,42 @@ InstallMethod(InversesOfSemigroupElementNC,
 "for a semigroup and a multiplicative element",
 [IsSemigroup, IsMultiplicativeElement],
 function(S, x)
-  if not IsFinite(S) then
-    TryNextMethod();
-  fi;
-  return Filtered(EnumeratorSorted(S), y -> x * y * x = x and y * x * y = y);
+    if not IsFinite(S) then
+        TryNextMethod();
+    fi;
+    return Filtered(EnumeratorSorted(S), 
+    y -> x * y * x = x and y * x * y = y);
+end);
+
+InstallMethod(InversesOfSemigroupElementNC,
+"for CanUseFroidurePin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],
+function(S, a)
+  local R, L, inverses, e, f, s;
+  R := RClass(S, a);
+  L := LClass(S, a);
+  inverses := [];
+
+  for e in Idempotents(R) do
+    s := RightGreensMultiplierNC(S, a, e) * e;
+    for f in Idempotents(L) do
+      Add(inverses, f * s);
+    od;
+  od;
+  return inverses;
+end);
+
+InstallMethod(OneInverseOfSemigroupElementNC,
+"for CanUseFroidurePin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],
+function(S, a)
+  local R, L, e, f, s;
+  R := RClass(S, a);
+  L := LClass(S, a);
+  e := Idempotents(R)[1];
```

Comment:
```suggestion
  e := Idempotents(R);
  if IsEmpty(e) then
    return fail;
   fi;
   e := e[1];
```

Reply:
Also maybe add a comment to the code that indicates that it might be better if we could obtain a single idempotent from `R` and `L` rather than all the idempotents.